tests/db_tests/test_queries.py

```diff
@@ -481,3 +484,27 @@ def test_project_if_no_outcomes(seeded_test_client, additional_test_data):
 
     test_df = pd.read_sql(test_query_proj.statement, con=db.engine.connect())
     assert list(test_df["project_name"])
+
+
+def test_with_transaction_max_retries(mocker, test_client):
```

Comment:
could we also add a test that retries, but then succeeds on a subsequent try? I think you can pass a list of exceptions to the param for side_effects to make a call to the same function do different behaviour each time. 

So basically, a test we can assert calls the "function" more than once, and whereby no exception is raised at the end (since on the 2nd or 3rd attempt it did not raise anything)